src/Sphere.jl

```diff
@@ -158,19 +158,20 @@ function normal_tvector_distribution(S::Sphere, x, σ)
 end
 
 @doc doc"""
-    mean(S::Sphere, x::AbstractVector, w::AbstractWeights; shuffle_rng=nothing, kwargs...)
+    mean(S::Sphere, x::AbstractVector[, w::AbstractWeights]; shuffle_rng=nothing, kwargs...)
 
-Compute the Riemannian mean of `x` using [`GeodesicInterpolationMethod`](@ref).
-If any `x` are not within $\frac{\pi}{2}$ of the estimated mean, then the
-estimate is used as the initial guess for the gradient method.
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of `x` using
+[`GeodesicInterpolation`](@ref). If any `x` are not within $\frac{\pi}{2}$
+of the estimated mean, then the estimate is used to initialize mean computation
+using the [`GradientDescent`](@ref).
 """
 mean(::Sphere, args...)
 
-function mean!(S::Sphere, y, x::AbstractVector, w::AbstractWeights; shuffle_rng = nothing, kwargs...)
-    mean!(S, y, x, w, GeodesicInterpolationMethod(); shuffle_rng = shuffle_rng, kwargs...)
+function mean!(S::Sphere, y, x::AbstractVector, w::AbstractVector; shuffle_rng = nothing, kwargs...)
+    mean!(S, y, x, w, GeodesicInterpolation(); shuffle_rng = shuffle_rng, kwargs...)
     for i in eachindex(x)
         @inbounds if distance(S, y, x[i]) ≥ π/2
```

Comment:
The question to a common implementation using a injectivity_radius instead of the magic number (pi/2) also applies here.

Reply:
The "magic number" is unique to each manifold and is more conservative than the injectivity radius. If enough of the samples fall outside of that ball, then the accuracy drops quite a bit for some manifolds, so we don't want to use injectivity radius here.

But I do agree a common implementation is good. I'll add a `GeodesicInterpolationWithinRadius` method that implements this with fallback and have the manifolds with finite radii default to that.

Reply:
With "magic number" I meant exactly that – to me (see the Sphere) that looked like the injectivity radius, that's why I would like to have a function for that. As a conservative idea I would be fine with a default of injectivity_radius/2, but I would like to have a name not a number here. I will check you update later, then :)